code-reviews/recipes/azure-pipelines-yaml.md

```diff
@@ -0,0 +1,119 @@
+
+Before starting review please see the following documents:
+
+- Make sure to check [Azure Pipelines YAML documentation](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema).
+
+- Understand how pipelines work [Pipeline run sequence](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/runs?view=azure-devops)
+
+- Get familiar with [Key concepts for new Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started/key-pipelines-concepts?view=azure-devops)
+
+  **Key concepts overview**
+![Azure Pipelines key concepts](images/key-concepts-overview.png)
+
+  - A trigger tells a Pipeline to run.
+  - A pipeline is made up of one or more stages. A pipeline can deploy to one or more environments.
+  - A stage is a way of organizing jobs in a pipeline and each stage can have one or more jobs.
+  - Each job runs on one agent. A job can also be agentless.
+  - Each agent runs a job that contains one or more steps.
+  - A step can be a task or script and is the smallest building block of a pipeline.
+  - A task is a pre-packaged script that performs an action, such as invoking a REST API or publishing a build artifact.
+  - An artifact is a collection of files or packages published by a run.
+
+## Code Review Checklist
+
+In addition to the [Code Review Checklist](../process-guidance/reviewer-guidance.md) you should also look for these Azure Pipelines YAML specific code review items.
+
+### Pipeline Structure
+
+- [ ] First understand [Pipeline structure](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#pipeline-structure) and identify components
+- [ ] Check the pipeline in Azure Pipelines
+- [ ] Edit the pipeline in Azure Pipelines and find root file
+- [ ] Find templates called in the pipeline
+- [ ] Check all the template file references to ensure it does not cause breaking changes, changing one file may effect multiple pipelines
+- [ ] Avoid long inline scipts in YAML file, consider to separate into a script file
+
+### YAML Structure
+
+- [ ] Avoid big templates, check and identify re-usable components
+- [ ] Separate environment specific variables
+  - [ ] Consider using variable templates per environment
+  - [ ] Or consider using variable group per environment in Azure Pipelines Library
+- [ ] Consider changes of values during `Queue Time`, `Compile Time` and `Runtime`
+- [ ] Check variable syntax values used with `Macro Syntax`, `Template Expression Syntax` and `Runtime Expression Syntax`
+- [ ] Is there any unused variable or parameter? Make sure those are removed in pipeline.
+- [ ] Does the pipeline meet with stage/job `Conditions` criterias?
+- [ ] Consider value changes: Variables can change during the pipeline, Parameters cannot
+
+### Permission Check & Security
+
+- [ ] Make sure pipeline is not printing any secret value in script
```

Comment:
Same here - turn the todo items into check items if you want to use a checklist

Reply:
Thanks Tess, reworded check list items.